scripts/gen_airbyte_classes.py

```diff
@@ -0,0 +1,579 @@
+KEYWORDS = ["None", "Any", "Union", "List", "Dict", "Optional", "T"]
+
+
+def _capitalize(s: str) -> str:
+    return s[:1].upper() + s[1:]
+
+
+def _to_class_name(name: str) -> str:
+    class_name = "".join([_capitalize(_remove_invalid_chars(x)) for x in re.split("\W+", name)])
+    if class_name in KEYWORDS:
+        class_name += "_"
+    return class_name
+
+
+TYPE_MAPPING = {
+    "string": "str",
+    "integer": "int",
+    "number": "float",
+    "boolean": "bool",
+}
+
+
+CHECK_MAPPING = {
+    "str": "check.str_param({}, '{}')",
+    "int": "check.int_param({}, '{}')",
+    "float": "check.float_param({}, '{}')",
+    "bool": "check.bool_param({}, '{}')",
+}
+
+
+class SchemaType(ABC):
+    """
+    Corresponds to the Python type of a field in a schema, has methods to generate
+    the Python code to annotate that type or check it at runtime.
+    """
+
+    @abstractmethod
+    def get_check(self, name: str, scope: Optional[str] = None):
+        """
+        Returns the dagster._check check for this type, e.g. check.str_param(name, 'name')
+        """
+
+    @abstractmethod
+    def annotation(self, scope: Optional[str] = None, quote: bool = False):
+        """
+        Returns the Python type annotation for this type, e.g. str or Union[str, int]
+        """
+
+    @property
+    def const_value(self):
+        """
+        If this is a constant field, returns the constant value, otherwise returns None
+        """
+        return None
+
+
+class RawType(SchemaType):
+    def __init__(self, schema_type_str: str, const_value: Optional[Any] = None):
+        if schema_type_str in TYPE_MAPPING:
+            self.type_str = TYPE_MAPPING[schema_type_str]
+        else:
+            self.type_str = schema_type_str
+        self._const_value = const_value
+
+    def __str__(self):
+        return self.type_str
+
+    @property
+    def const_value(self):
+        return self._const_value
+
+    def annotation(self, scope: Optional[str] = None, quote: bool = False):
+        if self.type_str in CHECK_MAPPING:
+            return self.type_str
+        scope = f"{scope}." if scope else ""
+
+        if quote:
+            return f'"{scope}{self.type_str}"'
+        return f"{scope}{self.type_str}"
+
+    def get_check(self, name: str, scope: Optional[str] = None):
+        if self.type_str in CHECK_MAPPING:
+            return CHECK_MAPPING[self.type_str].format(name, name)
+        scope = f"{scope}." if scope else ""
+        return "check.inst_param({}, '{}', {}{})".format(name, name, scope, self.type_str)
+
+
+class OptType(SchemaType):
+    def __init__(self, inner: SchemaType):
+        self.inner = inner
+
+    def __str__(self):
+        return f"Optional[{self.inner}]"
+
+    def annotation(self, scope: Optional[str] = None, quote: bool = False):
+        return f"Optional[{self.inner.annotation(scope, quote)}] = None"
+
+    def get_check(self, name: str, scope: Optional[str] = None):
```

Comment:
So my one main point of concern here (which is possibly fine?) is that `check.opt_list_param` will return an empty list when given `None` as an input, and I'm not sure we can be confident that all Airbyte sources / destinations will treat empty lists identically to `None` (I'd be surprised if they do)

Instead, we should use `check.opt_nullable_list_param` in those cases (mildly annoying because there's no analogous (e.g.) `check.opt_nullable_int_param`, so this logic will have to inspect the type of inner).

Reply:
Good call, added some logic to inspect the inner type and return a nullable check instead.